tests/unit/test_container_encoders.py

```diff
@@ -207,8 +207,10 @@ def test_screen_with_plate_encoder(self, screen_with_plates):
                     'Type': 'the_type',
                     'Color': -16777216,
                     'omero:status': 'the_status',
+                    'plate': 5,
```

Comment:
Chiming in on the discussion about the way to encode these backreferences, I would think something along the lines of

```
'parent': {
  '@type': '%s#Plate' % SPW_SCHEMA_URL,
  '@id: 5
},
```

might be a more generic representation and more JSON-LD compatible.
Note for deep hierarchies like plates, this approach will probably lead to similar issues as https://github.com/ome/omero-marshal/issues/22.

Reply:
I don't think this is really about "backreferences" to a "parent", but about not ignoring the IDs to other objects that we could encode with zero cost. So that encoding isn't lossy.
E.g. Image has IDs for `fileset`, `format`, `imagingenvironment`, `instrument`, `objectivesettings` etc, which are not encoded, even though this could be useful info to have in the JSON.

Reply:
Sorry for having dropped the ball on this. Seconding @sbesson, maybe the backreferences/parent terminology obscured the more important point. Looking at the rest of the output here, are there any examples of an object ID only being referenced as `"name": 123`? Since these may then form a loop, we're probably going to also need to think about what an "unloaded" object looks like.

Reply:
> are there any examples of an object ID only being referenced as "name": 123

Like `"plate": 5`?
I don't quite see how this may form a loop.

The whole graph traversal in `omero-metadata` is in one direction only. E.g. `Plate->Well->Annotation` or `Roi->Shape` and not the reverse, so I don't see where we'd get a loop.

For a Shape, even if the ROI is loaded, I wouldn't expect to encode it (as then you'd get recursive loops of encoding).
So it makes no difference if the ROI is loaded or not.